### PR TITLE
Adding interfaces for data access

### PR DESCRIPTION
**Pull request by ula0156**
This change brings:

1. Interfaces for access to data (products, stocks, reservations) necessary for switching to EntityFramework
2. Changed the in memory databases to comply with the new interfaces
3. Changed all the classes that need access to data to use the new interfaces
4. Reorganized the classes in namespaces
5. Renamed data access classes in order to be consistent

Ran application and verified basic functionality.

### CODE REVIEW

File: onlineShop/pages/SearchPage.cs [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-using OnlineShop.Products;
+using onlineShop.ProductPickers;
+using onlineShop.Products;
 using System.Collections.Generic;
 using System.Text;
 
-namespace OnlineShop.Pages
+namespace onlineShop.Pages
 {
     class SearchPage : IPage
```
Comment (simionliviumihai) [inline onlineShop/pages/SearchPage.cs]: This is not public as the other pages are. Please make it public.

File: onlineShop/Cart/Cart.cs [hunk 2]
```diff
@@ -1,12 +1,13 @@
-﻿using OnlineShop.Data;
-using OnlineShop.Products;
-using OnlineShop.Reservations;
+﻿using onlineShop.Data;
```
Comment (simionliviumihai) [inline onlineShop/Cart/Cart.cs:1]: Remove unused using in all changed files.

File: onlineShop/Cart/Cart.cs [hunk 3]
```diff
@@ -15,13 +16,13 @@ public class Cart
         // That's why we have to give a cart access to the inventory.
         // One productID might have several reservations -> list of reservations.
         private Dictionary<Guid, List<Reservation>> _reservations; // Guid - productId
-        private ReservationManager _reservationManager;
-        private ProductsDescriptions _productsDescriptions;
+        private ReservationsManager _reservationsManager;
+        private IProductsReader _inventory;
 
-        public Cart(ProductsDescriptions inventory, ReservationManager reservationManager)
+        public Cart(IProductsReader inventory, ReservationsManager reservationsManager)
```
Comment (simionliviumihai) [inline onlineShop/Cart/Cart.cs]: Rename inventory to productsReader.

File: onlineShop/Data/IStocksReader.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace onlineShop.Data
+{
+    public interface IStocksReader
```
Comment (simionliviumihai) [inline onlineShop/Data/IStocksReader.cs]: Add comment to interface.

File: onlineShop/Data/InMemory/InMemoryProductsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+﻿using onlineShop.Products;
+using System;
+using System.Collections.Generic;
+
+namespace onlineShop.Data.InMemory
+{
+    public class InMemoryProductsRepository
+    {
+        // Guid - product id, product, int - number of copies
```
Comment (simionliviumihai) [inline onlineShop/Data/InMemory/InMemoryProductsRepository.cs]: Update comment, I don't see any int - number of copies in the code

File: onlineShop/Data/InMemory/InMemoryStocksProvider.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using onlineShop.Data.InMemory;
+
+namespace onlineShop.Data.InMemory
+{
+    public class InMemoryStocksProvider : IStocksProvider
+    {
+        private InMemoryStocksRepository _productsStocks;
+
+        public InMemoryStocksProvider(InMemoryStocksRepository productsStocks)
```
Comment (simionliviumihai) [inline onlineShop/Data/InMemory/InMemoryStocksProvider.cs]: Rename productsStocks to stocksRepository

**Review by simionliviumihai — CHANGES_REQUESTED** [review]

File: onlineShop/pages/SearchPage.cs (hunk 1 above)
Comment (ula0156) [reply onlineShop/pages/SearchPage.cs]: Done

File: onlineShop/Cart/Cart.cs (hunk 2 above)
Comment (ula0156) [reply onlineShop/Cart/Cart.cs:1]: All of them are used

File: onlineShop/Data/IStocksReader.cs (hunk 4 above)
Comment (ula0156) [reply onlineShop/Data/IStocksReader.cs]: Done

File: onlineShop/Cart/Cart.cs (hunk 3 above)
Comment (ula0156) [reply onlineShop/Cart/Cart.cs]: Done

File: onlineShop/Data/InMemory/InMemoryProductsRepository.cs (hunk 5 above)
Comment (ula0156) [reply onlineShop/Data/InMemory/InMemoryProductsRepository.cs]: Done

File: onlineShop/Data/InMemory/InMemoryStocksProvider.cs (hunk 6 above)
Comment (ula0156) [reply onlineShop/Data/InMemory/InMemoryStocksProvider.cs]: Done

**Review by simionliviumihai — APPROVED** [review]

**Pull request merged**